src/runtime/cuda.cpp

```diff
@@ -533,29 +534,31 @@ WEAK int halide_cuda_device_malloc(void *user_context, buffer_t *buf) {
         return ctx.error;
     }
 
-    size_t size = buf_size(buf);
+    size_t size = buf->size_in_bytes();
     halide_assert(user_context, size != 0);
-    if (buf->dev) {
+    if (buf->device) {
         // This buffer already has a device allocation
         halide_assert(user_context, validate_device_pointer(user_context, buf, size));
         return 0;
     }
 
-    halide_assert(user_context, buf->stride[0] >= 0 && buf->stride[1] >= 0 &&
-                                buf->stride[2] >= 0 && buf->stride[3] >= 0);
+    // Check all strides positive.
+    for (int i = 0; i < buf->dimensions; i++) {
+        halide_assert(user_context, buf->dim[i].stride >= 0);
+    }
 
     debug(user_context) << "    allocating buffer of " << (uint64_t)size << " bytes, "
```

Comment:
Maybe also print out buf->dimensions so we know if there are more than 4. Or update dumping of the extents/strides to dump all dimensions.